app/repositories/AssessmentScoresRepository.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import factories.DateTimeFactory
+import model.UniqueIdentifier
+import model.assessmentscores._
+import reactivemongo.api.{ DB, ReadPreference }
+import reactivemongo.bson.{ BSONDocument, BSONObjectID, _ }
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait AssessmentScoresRepository {
+  def save(scoresAndFeedback: AssessmentScoresAllExercises): Future[Unit]
+
+  def find(applicationId: UniqueIdentifier): Future[Option[AssessmentScoresAllExercises]]
+
+  def findAll: Future[List[AssessmentScoresAllExercises]]
+}
+
+class AssessmentScoresMongoRepository(dateTime: DateTimeFactory)(implicit mongo: () => DB)
+  extends ReactiveRepository[AssessmentScoresAllExercises, BSONObjectID](CollectionNames.ASSESSMENT_SCORES, mongo,
+    AssessmentScoresAllExercises.jsonFormat, ReactiveMongoFormats.objectIdFormats)
+    with AssessmentScoresRepository with BaseBSONReader with ReactiveRepositoryHelpers {
+
+  // TODO: This save method does not remove exercise subdocument when in the case class they are None
```

Comment:
Given that I will continue working in this area later on, it is a good reminder to me.